test/python/transpiler/test_commuting2q_router.py

```diff
@@ -0,0 +1,74 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Tests for routing Commuting2qGateGrouper and Commuting2qGateRouter."""
+
+from qiskit.circuit import QuantumCircuit
+from qiskit.transpiler import PassManager, CouplingMap
+from qiskit.circuit.library import TwoLocal
+from qiskit.transpiler.passes import Commuting2qGateGrouper, Commuting2qGateRouter
+from qiskit.transpiler.passes.routing.commuting_2q_gate_routing import SwapStrategy
+
+from qiskit.test import QiskitTestCase
+
+
+class TestCommuting2qGateGrouperRouting(QiskitTestCase):
```

Comment:
I feel like the tests here could be a bit more thorough and targetted to just the new `Commuting2qGateGrouper` pass's behavior. Like it'd be good to test what the pass does with an empty dag, dags that have no 2q gates, that commuting gates are correctly grouped, etc. having the full path tests are good but I think we should test things in isolation on the new feature here too.